odoo/addons/base/models/ir_model.py

```diff
@@ -583,6 +583,12 @@ def _check_name(self):
                 msg = _("Field names can only contain characters, digits and underscores (up to 63).")
                 raise ValidationError(msg)
 
+    @api.constrains('size')
+    def _check_size(self):
+        for rec in self:
+            if rec.translate and rec.size > 0:
+                raise ValidationError(_("Translatable field should have Size 0"))
```

Comment:
I would use another sentence.
Something like "Translatable field cannot have a size".

Also, don't forget to add it in the `.pot` file.

Reply:
Changes are done!!